pkg/machine/hyperv/machine.go

```diff
@@ -601,18 +602,33 @@ func (m *HyperVMachine) Start(name string, opts machine.StartOptions) error {
 	// set starting back false now that we are running
 	m.Starting = false
 
-	if err := m.startShares(); err != nil {
-		return err
-	}
-
 	if m.HostUser.Modified {
 		if machine.UpdatePodmanDockerSockService(m, name, m.UID, m.Rootful) == nil {
 			// Reset modification state if there are no errors, otherwise ignore errors
 			// which are already logged
 			m.HostUser.Modified = false
 		}
 	}
+
 	// Write the config with updated starting status and hostuser modification
+	if err := m.writeConfig(); err != nil {
+		return err
+	}
+
+	// Check if gvproxy is still running.
+	// Do this *after* we write config, so we have still recorded that the
+	// VM is actually running - to ensure that stopping the machine works as
+	// expected.
+	_, err = psutil.NewProcess(gvproxyPid)
+	if err != nil {
+		return fmt.Errorf("gvproxy appears to have stopped (PID %d): %w", gvproxyPid, err)
+	}
+
+	// Finalize starting shares after we are confident gvproxy is still alive.
+	if err := m.startShares(); err != nil {
+		return err
+	}
+
 	return m.writeConfig()
```

Comment:
sorry didn't notice this one befofore, is there actually anything that needs to be written there?
startshares() seems to only use ssh and does not modify the config file so the last write would be unnecessary.

Reply:
the writeconfig is for writing the new machine state into the config file ... at question might be more along the lines of is the `m.Starting = false`  in the correct place regarding timing?

Reply:
We can probably drop it given the extra writeConfig() before for starting. I'll make the change.

Reply:
Done, repushed